ui/src/frontend/notes_panel.ts

```diff
@@ -311,6 +315,18 @@ export class NotesPanel extends Panel {
     globals.makeSelection(Actions.addNote({timestamp, color}));
   }
 
+  private onRightClick(x: number, _: number) {
+    if (x < 0) return;
+    for (const note of Object.values(globals.state.notes)) {
+      if (this.hoveredX && this.mouseOverNote(this.hoveredX, note)) {
```

Comment:
Assuming `this.hoveredX` is a constant in the context of this function, we should just have a check at the beginning
```
if (!this.hoveredX) return;
```

I.e., no need to iterate when we know all the iterations are going to no-op. I get that this just follows what was done in `onClick`, but that doesn't mean we have to repeat mistakes of the past (if my assumption above is correct).